src/map/ai/states/range_state.cpp

```diff
@@ -103,8 +107,21 @@ bool CRangeState::Update(time_point tick)
     {
         auto* PTarget = m_PEntity->IsValidTarget(m_targid, TARGET_ENEMY, m_errorMsg);
 
+        // Ensure player is using a weapon with the same delay and damage as when state was started
+        // This prevents players from starting a RA with short delay x-bow and ending with high dmg longbow
+        if (tick > GetEntryTime())
+        {
+            if (m_initialDamage != m_PEntity->GetRangedWeaponDmg() || m_initialDelay != m_PEntity->GetRangedWeaponDelay(false))
+            {
+                if (auto PChar = dynamic_cast<CCharEntity*>(m_PEntity))
+                {
+                    m_errorMsg = std::make_unique<CMessageBasicPacket>(PChar, PChar, 0, 0, MSGBASIC_NO_RANGED_WEAPON);
+                }
+            }
+        }
+
         CanUseRangedAttack(PTarget, true);
-        if (m_startPos.x != m_PEntity->loc.p.x || m_startPos.y != m_PEntity->loc.p.y)
+        if (HasMoved())
```

Comment:
👍

Reply:
guessing someone copied from magic casting logic. because you can star a magic cast run all over creation and as long as you manage to stop at the spot you started before casting completes, it'll go off 😂 have never heard the same about ranged and I haven't tried because I am not good enough at stopping in the right spot for that